jdbc/src/java/org/apache/hive/jdbc/HiveConnection.java

```diff
@@ -850,7 +851,7 @@ private String getJWTStringFromSession() {
    * @return TTransport
    * @throws TTransportException
    */
-  private TTransport createUnderlyingTransport() throws TTransportException {
+  private TTransport createUnderlyingTransport(int maxMessageSize) throws TTransportException {
     TTransport transport = null;
```

Comment:
nit: we can get this `maxMessageSize` through `getMaxMessageSize()` in the method instead by passing an argument.

Reply:
done